itlwm/hal_iwn/ItlIwn.cpp

```diff
@@ -4102,7 +4102,7 @@ iwn_set_link_quality(struct iwn_softc *sc, struct ieee80211_node *ni)
             }
         }
         
-        if (tab == 0)
+        if (tab == 0 && rflags == 0)
```

Comment:
This line is fixing the bug.

When `rflags` contains `IWN_RFLAG_MCS`, 0 is a good value for `tab` means mcs 0. In this case it should not break the loop. The result of this bug that the initial link quality retry table contain only legacy rates and no ht rate. In this condition the wifi card will not send any packets using ht rate, and will not trigger the `iwn_ra_choose` to update the table again. For some reason it was working with `mira`, but not the new `ra`. The fix is to check and break the loop when both `tab` and `rflags` are 0.

This bug was introduced in the df94da897c677a76b29a2df1d53f7515472e7b78 when port the logic from `iwm` to support tx mimo in `iwn`. The struct is one uint16 in `iwm` instead of two uint8 in `iwn`. The additional check is missed.